packages/pages/src/vite-plugin/modules/plugin.ts

```diff
@@ -0,0 +1,273 @@
+// The following code is added during build and removed after build is completed.
+const moduleContainerForBuildUseOnly = document.getElementById('${containerName}');
+if (!moduleContainerForBuildUseOnly) {
+  throw new Error('could not find ${containerName} element');
+}
+
+ReactDOM.render(
+  <React.StrictMode>
+    <${moduleName}/>
+  </React.StrictMode>,
+  moduleContainerForBuildUseOnly
+);
+  `;
+};
+
+const moduleResponseHeaderProps = {
+  headerKey: "Access-Control-Allow-Origin",
+  headerValues: ["*"],
+};
+
+type FileInfo = {
+  path: string;
+  name: string;
+};
+
+export const buildModules = async (
+  projectStructure: ProjectStructure
+): Promise<void> => {
+  if (!shouldBundleModules(projectStructure)) {
+    return;
+  }
+
+  const { rootFolders, subfolders, envVarConfig } = projectStructure.config;
+  const outdir = path.join(rootFolders.dist, subfolders.modules);
+
+  const filepaths: { [s: string]: FileInfo } = {};
+  glob
+    .sync(
+      convertToPosixPath(
+        path.join(rootFolders.source, subfolders.modules, "**/*.{jsx,tsx}")
+      ),
+      { nodir: true }
+    )
+    .forEach((f) => {
+      const filepath = path.resolve(f);
+      const moduleName = getModuleName(filepath);
+      const { name } = path.parse(filepath);
+      filepaths[moduleName ?? name] = { path: filepath, name: name };
+    });
+
+  const logger = createLogger();
+  const loggerInfo = logger.info;
+
+  if (tailwindBaseExists()) {
+    // TODO add link to recommended implementation for user.
+    logWarning(
+      `Please be aware that using @tailwind base applies styles globally. This can affect code outside of the widget.`
+    );
+  }
+
+  for (const [moduleName, fileInfo] of Object.entries(filepaths)) {
+    const index = addExtraModuleCode(fileInfo.path, moduleName);
+    logger.info = (msg, options) => {
+      if (msg.includes("building for production")) {
+        loggerInfo(pc.green(`\nBuilding ${moduleName} module...`));
+        return;
+      }
+      loggerInfo(msg, options);
+    };
+
+    // For each module, add response header to config.yaml.
+    // Users can manually adjust headerKey and headerValue in their config.yaml.
+    // As long as pathPattern matches, it won't be overwitten.
+    addResponseHeadersToConfigYaml(
+      projectStructure,
+      {
+        pathPattern: `^modules/${moduleName}.*`,
+        ...moduleResponseHeaderProps,
+      },
+      "# The ^modules/ header allows access to your modules from other sites\n"
+    );
+
+    await build({
+      customLogger: logger,
+      configFile: false,
+      envDir: envVarConfig.envVarDir,
+      envPrefix: envVarConfig.envVarPrefix,
+      resolve: {
+        conditions: ["worker", "webworker"],
+      },
+      publicDir: false,
+      css: {
+        postcss: getPostCssConfigFilepath(
+          rootFolders,
+          subfolders,
+          fileInfo.name
+        ),
+      },
+      esbuild: {
+        logOverride: {
+          "css-syntax-error": "silent",
```

Comment:
I don't think we want this. Also esbuild shouldn't be used at all so I don't think we need the whole esbuild block.

Reply:
Was trying to suppress this warning: 
```
warnings when minifying css:
▲ [WARNING] Unexpected "{" [css-syntax-error]

    <stdin>:630:1:
      630 │  {
          ╵  ^
```
The exact line it is warning you about shifts, but is always there when minimize: true.

Reply:
Oh that is really weird. It did properly suppress the warning when running npm run prod

Reply:
https://github.com/vitejs/vite/issues/9196
https://github.com/vitejs/vite/pull/9210

Reply:
Can also add a catch error in the logger for this instead if we prefer.